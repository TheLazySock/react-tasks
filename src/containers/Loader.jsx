import React from 'react';

import CircularProgress from 'material-ui/CircularProgress';

const Loader = props => {
    return (
        <div className="loader">
            {
                props.loading
                ? <CircularProgress className="spinner" size={250} thickness={15} color={'#da395c'}/>
                : props.children
            }
        </div>
    );
}

export default Loader;