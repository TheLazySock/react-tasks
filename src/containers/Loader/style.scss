.loader {
    display: flex;
    flex-direction: column;
    width: 100%;
    .spinner {
        display: block;
        margin: 100px auto;
    }
}