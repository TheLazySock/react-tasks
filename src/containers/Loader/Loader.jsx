// @flow
import React from 'react';
import CircularProgress from 'material-ui/CircularProgress';
import style from './style.scss';

type LoaderProps = {
    loading: boolean,
    children: Array<any>
}

const Loader = (props: LoaderProps) => (
    (
        <div className={style.loader}>
            {
                props.loading
                    ? <CircularProgress className={style.spinner} size={250} thickness={15} color={'#da395c'}/>
                    : props.children
            }
        </div>
    )
);

export default Loader;
