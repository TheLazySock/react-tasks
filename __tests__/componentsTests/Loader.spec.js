import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../../src/containers/Loader';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('>>>LOADER tests', () => {    
    it('loader have to match snapshot with true loading', () => {
        const container = shallow(<Loader loading={true} children={[{id: 0}, {id: 1}]} />);
        expect(shallowToJson(container)).toMatchSnapshot();
        expect(container.children().length).toBe(1);
    });
    
    it('loader have to match snapshot with false loading', () => {
        const container = shallow(<Loader loading={false} children={[{id: 0}, {id: 1}]} />);
        expect(shallowToJson(container)).toMatchSnapshot();
        expect(container.children().length).toBe(2);
    });
});