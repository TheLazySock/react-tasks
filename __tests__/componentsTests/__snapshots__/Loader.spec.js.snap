// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>LOADER tests loader have to match snapshot with false loading 1`] = `<div />`;

exports[`>>>LOADER tests loader have to match snapshot with true loading 1`] = `
<div>
  <CircularProgress
    color="#da395c"
    max={100}
    min={0}
    mode="indeterminate"
    size={250}
    thickness={15}
    value={0}
  />
</div>
`;
